Add analysis of moran process

drvinceknight opened the pull request.

drvinceknight committed:
Add a notebook.

drvinceknight committed:
Plot figure.

drvinceknight committed:
Add discussion of Moran process.

drvinceknight committed:
Run black.

Nikoleta-v3 commented on main.tex:
```diff
@@ -500,6 +500,80 @@ \section{Results}\label{section:results}
+
+where:
+
+\[ \gamma_i = \frac{ p_{k, k - 1} }{ p_{k, k + 1} } \]
```
```suggestion
\[ \gamma_i = \frac{ p_{k, k - 1} }{ p_{k, k + 1} }. \]
```

Nikoleta-v3 commented on main.tex:
```diff
@@ -500,6 +500,80 @@ \section{Results}\label{section:results}
+In the case considered here \(k\) is the number of best response players in the
+population, so that \(N - k\) is the number of opponents.
+
+The difference proposed here is to allow the best response player to act
```
Different to what?

drvinceknight replied:
Yeah this paragraph needs to be improved it.

Nikoleta-v3 commented on main.tex:
```diff
@@ -500,6 +500,80 @@ \section{Results}\label{section:results}
+
+\[ \gamma_i = \frac{ p_{k, k - 1} }{ p_{k, k + 1} } \]
+
+In the case considered here \(k\) is the number of best response players in the
```
Just making sure: we do mean small k here right?

drvinceknight replied:
No we mean big `K` throughout.

Nikoleta-v3 commented on main.tex:
```diff
@@ -500,6 +500,80 @@ \section{Results}\label{section:results}
+\(A ^ {(k)}_{11}\) is the long run utility of the best response player against itself.
+\(A ^ {(k)}_{12}\) is the long run utility of the best response player against the opponent.
+\(A ^ {(k)}_{11}\) is the long run utility of the opponent against the best response player.
+\(A ^ {(k)}_{11}\) is the long run utility of the opponent against itself.
```
```suggestion

\begin{itemize}
\item \(A ^ {(k)}_{11}\) is the long run utility of the best response player against itself.
\item \(A ^ {(k)}_{12}\) is the long run utility of the best response player against the opponent.
\item \(A ^ {(k)}_{11}\) is the long run utility of the opponent against the best response player.
\item \(A ^ {(k)}_{11}\) is the long run utility of the opponent against itself.
\end{itemize}
```

Nikoleta-v3 commented on main.tex:
```diff
@@ -500,6 +500,80 @@ \section{Results}\label{section:results}
+\(A ^ {(k)}_{11}\) is the long run utility of the opponent against the best response player.
+\(A ^ {(k)}_{11}\) is the long run utility of the opponent against itself.
+
+The matrix \(A ^ {(k)}\) is calculated using (\ref{eq:optimisation_quadratic})
```
```suggestion
The matrix \(A ^ {(k)}\) is calculated using Eq. (\ref{eq:optimisation_quadratic})
```

Nikoleta-v3 commented on main.tex:
```diff
@@ -500,6 +500,80 @@ \section{Results}\label{section:results}
+\(A ^ {(k)}_{11}\) is the long run utility of the opponent against itself.
+
+The matrix \(A ^ {(k)}\) is calculated using (\ref{eq:optimisation_quadratic})
+for each value of \(k\) once the best response dynamics algorithm has calculated
```
Can we maybe describe this with a bit better

drvinceknight replied:
Yeah this can all be improved, we could have a picture of the Moran process and a small numeric example.

Nikoleta-v3 commented on main.tex:
```diff
@@ -500,6 +500,80 @@ \section{Results}\label{section:results}
+\[f_2^{(k)} = (k) A_{21}^{(k)} + (N - k - 1)A_{22}^{(k)}\]
+
+(\(f_1^{(k)}\) is the fitness of the best response player, and \(f_2^{(k)}\) is
+the fitness of the opponent)
```
```suggestion
where \(f_1^{(k)}\) is the fitness of the best response player, and \(f_2^{(k)}\) is
the fitness of the opponent.
```

Nikoleta-v3 commented on main.tex:
```diff
@@ -500,6 +500,80 @@ \section{Results}\label{section:results}
+experiments is given by % TODO Add mean
+and in some cases this dynamic updating results in a 25\% increase in the
+absorption probability.
+As denoted before it is clear that the best response strategy does not have a
```
long sentence. I think this paragraph needs a bit more. Maybe even explain the ratio here as well as in the caption.

drvinceknight replied:
Yeah we could possibly define some notation for the non dynamic case and explain that.

Nikoleta-v3 reviewed (commented):
I reviewed the addition on the Moran results.

Regarding the manuscript as a whole I believe it needs restructure and a bit more discussion on the conclusions of the Moran process.

drvinceknight commented on main.tex:
```diff
@@ -500,6 +500,80 @@ \section{Results}\label{section:results}
+\(A ^ {(k)}_{11}\) is the long run utility of the opponent against itself.
+
+The matrix \(A ^ {(k)}\) is calculated using (\ref{eq:optimisation_quadratic})
+for each value of \(k\) once the best response dynamics algorithm has calculated
```
Needs to be big `\(K\)`

drvinceknight committed:
Make minor suggestions.

Co-authored-by: Nikoleta Glynatsi <GlynatsiNE@cardiff.ac.uk>

drvinceknight committed:
Add `where`

Co-authored-by: Nikoleta Glynatsi <GlynatsiNE@cardiff.ac.uk>

drvinceknight committed:
Update results with new data.

drvinceknight commented:
268edc2 updates the figures with latest moran data (as of 2020-09-11).

Link to the figure: https://github.com/Nikoleta-v3/Memory-size-in-the-prisoners-dilemma/blob/268edc22e2c21b07e87b0153ab1594b789348e9a/img/dynamic_moran_process_results/main.pdf

Nikoleta-v3 committed:
uses capital K consistently over the manuscript

Nikoleta-v3 committed:
suggests structure

I am happy for the results section to remain as it is. I suggest
including some subsection will help with the narative.

The suggested titles are just suggestios.
I am happy to un-do the commit.
The moran section still needs tweaking (which includes adding discussion in the conclusion)

drvinceknight committed:
Reorder the results section.

This now goes:

S. Results

s. Best response
s. Best response against a group
s. Limitations of extortion
s. Limitations of memory
s. Dynamic best response player

drvinceknight committed:
Fix Moran process part of results section.

drvinceknight committed:
s/N/n in Moran process.

drvinceknight committed:
Adjust discussion.

Nikoleta-v3 committed:
removes discussion on self interactions

Nikoleta-v3 committed:
corrects minor stuff to compile document

Nikoleta-v3 commented:
I am going to go ahead an merge this `PR` so I can compile and review the manuscript with all the changes (including #77 ).

Nikoleta-v3 commented:
We can `PR` our final suggestions to `prep-manuscript-for-resubmission`
